In [1]:
import numpy as np 
from sklearn.model_selection import train_test_split
from sklearn.feature_extraction.text import CountVectorizer
from sklearn.naive_bayes import MultinomialNB
from sklearn.metrics import accuracy_score, precision_score, recall_score, classification_report
import pandas as pd

In [6]:
groceries_df = pd.read_csv("datasets/groceries/groceries.csv", sep="\n")
groceries_df

citrus fruit,semi-finished bread,margarine,ready soups
0                          tropical fruit,yogurt,coffee    
1                                            whole milk    
2            pip fruit,yogurt,cream cheese,meat spreads    
3     other vegetables,whole milk,condensed milk,lon...    
4        whole milk,butter,yogurt,rice,abrasive cleaner    
...                                                 ...    
9829  sausage,chicken,beef,hamburger meat,citrus fru...    
9830                                  cooking chocolate    
9831  chicken,citrus fruit,other vegetables,butter,y...    
9832  semi-finished bread,bottled water,soda,bottled...    
9833  chicken,tropical fruit,other vegetables,vinega...    

[9834 rows x 1 columns]

9834 samples of category "Grocery" in groceries.csv file


In [23]:
online_retail_df = pd.read_csv("datasets/Online Retail.csv")
online_retail_df = online_retail_df[['InvoiceNo', 'Description']] # use only the columns needed
online_retail_df

InvoiceNo                          Description
0         536365   WHITE HANGING HEART T-LIGHT HOLDER
1         536365                  WHITE METAL LANTERN
2         536365       CREAM CUPID HEARTS COAT HANGER
3         536365  KNITTED UNION FLAG HOT WATER BOTTLE
4         536365       RED WOOLLY HOTTIE WHITE HEART.
...          ...                                  ...
541904    581587          PACK OF 20 SPACEBOY NAPKINS
541905    581587         CHILDREN'S APRON DOLLY GIRL 
541906    581587        CHILDRENS CUTLERY DOLLY GIRL 
541907    581587      CHILDRENS CUTLERY CIRCUS PARADE
541908    581587        BAKING SET 9 PIECE RETROSPOT 

[541909 rows x 2 columns]

In [29]:
grpby_invoice = online_retail_df.groupby("InvoiceNo")
len(grpby_invoice.groups)

25900

In [25]:
first_invoice = grpby_invoice.get_group('538070')
first_invoice

InvoiceNo                    Description
20681    538070   COFFEE MUG DOG + BALL DESIGN
20682    538070   COLOURING PENCILS BROWN TUBE
20683    538070           BLACK KITCHEN SCALES
20684    538070         TWO DOOR CURIO CABINET
20685    538070  BLUE GIANT GARDEN THERMOMETER

In [26]:
print(type(first_invoice))

<class 'pandas.core.frame.DataFrame'>


In [28]:
first_invoice_string = first_invoice.Description.str.cat(sep=", ")
first_invoice_string

'COFFEE MUG DOG + BALL DESIGN, COLOURING PENCILS BROWN TUBE, BLACK KITCHEN SCALES, TWO DOOR CURIO CABINET, BLUE GIANT GARDEN THERMOMETER'

In [35]:
to_save = grpby_invoice.apply(lambda a1: a1["Description"].str.cat(sep=", "))

Now that we have 25900 samples in a pandas Series, save it to a file for fitting the model in the future

In [40]:
to_save_df = pd.DataFrame({'InvoiceNo': to_save.index, "Description": to_save.values})
#to_save_df.to_csv("online_retail.csv", index=False)
to_save_df

InvoiceNo                                        Description
0        536365  WHITE HANGING HEART T-LIGHT HOLDER, WHITE META...
1        536366  HAND WARMER UNION JACK, HAND WARMER RED POLKA DOT
2        536367  ASSORTED COLOUR BIRD ORNAMENT, POPPY'S PLAYHOU...
3        536368  JAM MAKING SET WITH JARS, RED COAT RACK PARIS ...
4        536369                           BATH BUILDING BLOCK WORD
...         ...                                                ...
25895   C581484                        PAPER CRAFT , LITTLE BIRDIE
25896   C581490  VICTORIAN GLASS HANGING T-LIGHT, ZINC T-LIGHT ...
25897   C581499                                             Manual
25898   C581568                         VICTORIAN SEWING BOX LARGE
25899   C581569  HANGING HEART JAR T-LIGHT HOLDER, 36 PENCILS T...

[25900 rows x 2 columns]